R/lists_subscriptions.R

```diff
@@ -31,64 +22,21 @@ lists_subscriptions <- function(user,
                                 cursor = "-1",
                                 parse = TRUE,
                                 token = NULL) {
-  if (n > 1000) {
-    n <- ceiling(n / 1000)
-    count <- 1000
-  } else {
-    count <- n
-    n <- 1L
-  }
-  args <- list(
-    user = user,
-    n = count,
-    cursor = cursor,
-    parse = parse,
-    token = token
-  )
-  out <- vector('list', n)
-  for (i in seq_along(out)) {
-    out[[i]] <- do.call("lists_subscriptions_call", args)
-    args$cursor <- attr(out[[i]], "next_cursor")
-    if (is.null(args$cursor) || identical(args$cursor, "0")) {
-      break
-    }
-  }
-  if (parse) {
-    out <- do_call_rbind(out)
-  }
-  out
-}
-
-
-lists_subscriptions_call <- function(user,
-                                     n = 20,
-                                     cursor = "-1",
-                                     parse = TRUE,
-                                     token = NULL) {
+  
   params <- list(
     count = n,
     cursor = cursor
   )
   params[[user_type(user)]] <- user
 
-  r <- TWIT_get(token, "/1.1/lists/subscriptions", params)
-
-  ## get/set next_cursor value
-  if (has_name_(r, "next_cursor_str")) {
-    next_cursor <- r[["next_cursor_str"]]
-  } else {
-    next_cursor <- "0"
-  }
-
-  ## parse to data frame
-  if (parse && has_name_(r, "lists")) {
-    r <- as_lists_subscriptions(r$lists)
-    r <- as.data.frame(r)
+  r <- TWIT_paginate_cursor(token, "/1.1/lists/subscriptions", params,
+    page_size = 20,
```

Comment:
Page_size limit is 1000 according to the documentation
```suggestion
    page_size = 1000,
```